File: C++/AR_140.cpp
#include <iostream>
#include <iomanip>
#include <fstream>
#include <vector>
#include <assert.h>
#include <stdio.h>
#include <algorithm>
using namespace std;


int sumDigitos(int numero) {
	int ret;

	if (numero / 10 == 0) {
		ret = numero % 10;
		cout << numero;
	}
	else {
		ret = sumDigitos(numero / 10) + (numero % 10);
		cout << " + ";
		cout << (numero % 10);
	}
	return ret;
}


bool solve() {
	int n;
	cin >> n;
	if (n < 0)
		return false;

	int suma = sumDigitos(n);
	cout << " = " << suma << '\n';

	return true;

}

int main() {
	while (solve());

	return 0;
}

File: Java/AR_140.java
import java.util.Scanner;

public class AR_140 {
    static Scanner scanner=new Scanner(System.in);

    public static int sumDigitos(int numero) {
        int ret;

        if (numero/10==0) {
            ret=numero%10;
            System.out.print(numero);
        }
        else {
            ret=sumDigitos(numero/10)+(numero%10);
            System.out.print(" + "+(numero%10));
        }
        
        return ret;
    }

    public static boolean solve() {
        if (!scanner.hasNextInt()) return false;
        
        int n=scanner.nextInt();
        if (n<0) return false;

        int suma=sumDigitos(n);
        System.out.println(" = "+suma);

        return true;
    }

    public static void main(String[] args) {
        while (solve());

        scanner.close();
    }
}